api/stream.go

```diff
@@ -315,6 +333,26 @@ func (r streamRoutes) deleteVideoSection(c *gin.Context) {
 	if err != nil {
 		log.WithError(err).Error("Can't delete video-section")
 	}
+
+	sections, err := r.VideoSectionDao.GetByStreamId(context.Stream.ID)
+	if err != nil {
+		log.WithError(err).Error("failed to create video sections")
+		c.AbortWithStatus(http.StatusInternalServerError)
+	}
+
+	go func() {
+		parameters := generateVideoSectionImagesParameters{
+			sections:           sections,
+			playlistUrl:        context.Stream.PlaylistUrl,
+			courseName:         context.Course.Name,
+			courseTeachingTerm: context.Course.TeachingTerm,
+			courseYear:         uint32(context.Course.Year),
+		}
+		err := GenerateVideoSectionImages(r.DaoWrapper, &parameters)
+		if err != nil {
+			log.WithError(err).Error("failed to generate video section images")
```

Comment:
Why do we need to regenerate images when a section is deleted?

Reply:
Added a `DeleteSectionImage` request which solves that problem.